pkg/knit/cmd/k8s/podinfo.go

```diff
@@ -0,0 +1,204 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ */
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/user"
+	"path/filepath"
+	"text/template"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/openshift-kni/debug-tools/pkg/knit/cmd"
+	"github.com/spf13/cobra"
+)
+
+type podInfoOptions struct {
+	nodeName string
+}
+
+//Only need some info about the pod.
+// Right now is:
+// - pod name
+// - pod namespace
+// - node name
+// - status.qosClass
+// - containers
+//   - requests cpu
+//   - limits cpu
+// Note this output format could change but it would be parsed on insight rules
+// so the change should be sync with it.
+// Caution: We filter the data from pods to avoid exposing sensible information
+// (like environment variables or input parameters which can contain passwords)
+// so take care of that when changing this template.
+const default_template string = `
+[
+	{{- range $idx, $item := .Items}}
+	{{- if (ne $idx 0)}},{{end}}
+	{
+		"namespace":"{{.ObjectMeta.Namespace}}", 
+		"name":"{{.ObjectMeta.Name}}", 
+		"nodeName":"{{.Spec.NodeName}}", 
+		"qosClass": "{{.Status.QOSClass}}", 
+		{{- if .Spec.Containers }}
+		"containers": [
+		{{- range $cdx, $cont := .Spec.Containers -}}
+			{{- if (ne $cdx 0) }},{{ end }} 
+			{
+				"name":"{{.Name}}"
+				{{- if or .Resources.Requests .Resources.Limits -}}
+				,
+				"resources": {
+					{{- if .Resources.Limits}} {{if .Resources.Limits.Cpu}} 
+					"limits": {
+						"cpu": "{{.Resources.Limits.Cpu}}"
+					}
+					{{- end }}{{end}}
+					{{- if .Resources.Requests}}{{if .Resources.Requests.Cpu -}}
+					,
+					"requests": {
+						"cpu": "{{.Resources.Requests.Cpu}}"
+					}
+					{{- end }}{{end}}
+				} 
+				{{- end }} 
+			} 
+		{{- end }} 
+		]
+		{{- end }} 
+	}
+	{{- end }}
+]`
+
+func NewPodInfoCommand(knitOpts *cmd.KnitOptions) *cobra.Command {
+
+	opts := &podInfoOptions{}
+	podInfo := &cobra.Command{
+		Use:   "podinfo",
+		Short: "get pod status/spec information from cluster",
```

Comment:
non-blocking: maybe `get pod information complementing the podresources data`?
(or something like this)

Reply:
Ok I will revisit this